### PR TITLE
Structured suggestion for missing `mut`/`const` in raw pointer

### LINKED ISSUE #102261: Improve diagnostics for incorrect double pointer type
Given the following code: [link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=6ef90d137a70b94d60c117306b28f079)

```rust
fn main() {
    let x: i32 = 5;
    let ptr: *const i32 = &x;
    let dptr: **const i32 = &ptr;
}
```

The current output is:

```
   Compiling playground v0.0.1 (/playground)
error: expected mut or const in raw pointer type
 --> src/main.rs:4:15
  |
4 |     let dptr: **const i32 = &ptr;
  |               ^ expected mut or const in raw pointer type
  |
  = help: use `*mut T` or `*const T` as appropriate

error: could not compile `playground` due to previous error
```

Ideally the output should suggest correct ways to specify a double pointer type (e.g. `*const *const T`)(May be able to extend for more levels of indirection if wanted), instead of a generic single pointer suggestion, which may not be what the user wants.

### PR DESCRIPTION
**Pull request by compiler-errors**
Fixes #102261

### CODE REVIEW

File: compiler/rustc_parse/src/parser/ty.rs [hunk 1]
```diff
@@ -397,10 +397,14 @@ impl<'a> Parser<'a> {
     fn parse_ty_ptr(&mut self) -> PResult<'a, TyKind> {
         let mutbl = self.parse_const_or_mut().unwrap_or_else(|| {
             let span = self.prev_token.span;
-            let msg = "expected mut or const in raw pointer type";
+            let msg = "expected `mut` or `const` keyword in raw pointer type";
             self.struct_span_err(span, msg)
-                .span_label(span, msg)
-                .help("use `*mut T` or `*const T` as appropriate")
+                .span_suggestions(
```
Comment (compiler-errors) [inline compiler/rustc_parse/src/parser/ty.rs:402]: Do we support `span_suggestions` in structured subdiagnostics? If so, then I can put up another commit to convert this.

Comment (davidtwco) [reply compiler/rustc_parse/src/parser/ty.rs:402]: I don't think we do yet, I'll add that to #100717 as a task to do.

File: compiler/rustc_parse/src/parser/ty.rs [hunk 2]
```diff
@@ -397,10 +397,14 @@ impl<'a> Parser<'a> {
     fn parse_ty_ptr(&mut self) -> PResult<'a, TyKind> {
         let mutbl = self.parse_const_or_mut().unwrap_or_else(|| {
             let span = self.prev_token.span;
-            let msg = "expected mut or const in raw pointer type";
+            let msg = "expected `mut` or `const` keyword in raw pointer type";
             self.struct_span_err(span, msg)
```
Comment (davidtwco) [inline compiler/rustc_parse/src/parser/ty.rs:401]:
```suggestion
            self.struct_span_err(span, "expected `mut` or `const` keyword in raw pointer type")
```

**Review by davidtwco — APPROVED** [review]
One nit, then r=me

**Review by davidtwco — APPROVED** [review]

**Comment by compiler-errors** [comment]
Test failure looks unrelated?

@bors retry

**Pull request merged**